### PR TITLE
fix(scaledobject): reconcile hpa on label or annotations change

### PR DESCRIPTION
**Pull request by chubchubsancho**
_Provide a description of what has been changed_

HPA need to be reconcile on scaledObject annotations or labels update

### Checklist

- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #

#5468

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ Here is an overview of all new **experimental** features:
 
 ### Fixes
 
+- **General**: Fix reconcile HPA when ScaledObject annotations or labels is updated ([#5468](https://github.com/kedacore/keda/issues/5468))
```
Comment (jycamier) [inline CHANGELOG.md:69]:
```suggestion
- **General**: Fix reconcile HPA when ScaledObject annotations or labels are updated ([#5468](https://github.com/kedacore/keda/issues/5468))
```

Comment (chubchubsancho) [reply CHANGELOG.md:69]: nice catch @jycamier

File: controllers/keda/scaledobject_controller.go [hunk 2]
```diff
@@ -132,6 +132,8 @@ func (r *ScaledObjectReconciler) SetupWithManager(mgr ctrl.Manager, options cont
 				kedacontrollerutil.PausedReplicasPredicate{},
 				kedacontrollerutil.ScaleObjectReadyConditionPredicate{},
 				predicate.GenerationChangedPredicate{},
+				predicate.LabelChangedPredicate{},
+				predicate.AnnotationChangedPredicate{},
```
Comment (JorTurFer) [inline controllers/keda/scaledobject_controller.go:136]: Adding this predicate we don't need 
```golang
kedacontrollerutil.PausedPredicate{},
kedacontrollerutil.PausedReplicasPredicate{},
```
as both are allowing changes on annotations.

**Review by JorTurFer — COMMENTED** [review]
Looking good! only a small nit inline.
Although it's okey, I want to know @zroubalik's thoughts because maybe we weren't checking annotations and labels for any reason I'm missing

**Review by zroubalik — COMMENTED** [review]
The reason for not reconciling on labels/annotations was to limit the number of reconciliations in total and subsequent re-creation of scale loops.  I am not sure we should change the behavior this way. Maybe adding specific fields for annotations/labels in `spec.advanced.hpconfig..` ? Not sure though

**Comment by tomkerkhove** [comment]
@chubchubsancho Any update on this PR?

**Comment by chubchubsancho** [comment]
> @chubchubsancho Any update on this PR?

@tomkerkhove sorry i have not so much time past week to rework on this PR. I need to take a look at @zroubalik approach.

**Comment by zroubalik** [comment]
> > @chubchubsancho Any update on this PR?
> 
> @tomkerkhove sorry i have not so much time past week to rework on this PR. I need to take a look at @zroubalik approach.

no worries

**Comment by zroubalik** [comment]
There are still problems in checks, any update here please?

**Comment by chubchubsancho** [comment]
@zroubalik i've fixed problems

**Pull request closed without merging**